libcontainer/rootfs_linux.go

```diff
@@ -463,8 +479,35 @@ func mountToRootfs(m *configs.Mount, c *mountConfig) error {
 		if err := prepareBindMount(m, rootfs, mountFd); err != nil {
 			return err
 		}
-		if err := mountPropagate(m, rootfs, mountLabel, mountFd); err != nil {
-			return err
+
+		if m.IsBind() && m.IsIDMapped() {
+			if idmapFd == nil {
+				return fmt.Errorf("error creating mount %+v: idmapFd is nil, should point to a valid fd", m)
```

Comment:
Same here -- I don't like printing out the whole nine yards of mount structure in an error message. Perhaps it can be identified by its destination?

Reply:
Ok. Let's centralize the discussion on the other thread and then we apply what we agreed here too